saleor/graphql/product/mutations/channels.py

```diff
@@ -335,3 +346,76 @@ def perform_mutation(cls, _root, info, id, input):
         return ProductVariantChannelListingUpdate(
             variant=ChannelContext(node=variant, channel_slug=None)
         )
+
+
+class CollectionChannelListingUpdateInput(graphene.InputObjectType):
+    add_channels = graphene.List(
+        graphene.NonNull(PublishableChannelListingInput),
+        description="List of channels to which the collection should be assigned.",
+        required=False,
+    )
+    remove_channels = graphene.List(
+        graphene.NonNull(graphene.ID),
+        description="List of channels from which the collection should be unassigned.",
+        required=False,
+    )
+
+
+class CollectionChannelListingUpdate(BaseChannelListingMutation):
+    collection = graphene.Field(
+        Collection, description="An updated collection instance."
+    )
+
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of a collection to update.")
+        input = CollectionChannelListingUpdateInput(
+            required=True,
+            description=(
+                "Fields required to create or update collection channel listings."
+            ),
+        )
+
+    class Meta:
+        description = "Manage collection's availability in channels."
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        error_type_class = CollectionChannelListingError
+        error_type_field = "collection_channel_listing_errors"
+
+    @classmethod
+    def add_channels(cls, collection: "CollectionModel", add_channels: List[Dict]):
+        for add_channel in add_channels:
+            defaults = {
+                "is_published": add_channel.get("is_published"),
+                "publication_date": add_channel.get("publication_date", None),
+            }
```

Comment:
If we implement it in this way we can accidentally remove some data. Look in `add_channels` on `ProductChannelListingUpdate`.

In your solution, if you update only `is_published` defaults will contain `"publication_date": None`, that means you want to clear this field.